### PR TITLE
Adapt to new Hammer API

### PR DESCRIPTION
**Pull request by ruslandoga**
This PR conforms to the new api introduced in https://github.com/ExHammer/hammer/pull/104 and also improves performance somewhat by removing unnecessary `GenServer` processes and `poolboy`

### CODE REVIEW

File: .tool-versions
Comment (epinault) [inline .tool-versions:1]: again can we leave this one?

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,4 +1,11 @@
 # Changelog
+
+## 7.0.0 (unreleased)
+
+### Changed
+
+- Conform to new Hammer API
```
Comment (epinault) [inline CHANGELOG.md:7]:
```suggestion
- Conform to new Hammer API that ships in 7.x release
```

File: compose.yml
Comment (epinault) [inline compose.yml:1]: why renaming the docker-compose to compose? did the filename change? for both Docker or Lima/Colima work with the older file name by default

**Comment by epinault** [comment]
fine for an RC. for final I do want typespec. I have spent time adding them in the past to help documenting and dialyzer be more correct

File: compose.yml
Comment (ruslandoga) [reply compose.yml:1]: https://github.com/compose-spec/compose-spec/blob/main/03-compose-file.md

**Comment by ruslandoga** [comment]
The functions here are `@callback` implementations that already have typespecs.

File: .tool-versions
Comment (ruslandoga) [reply .tool-versions:1]: Sure.

File: compose.yml
Comment (epinault) [reply compose.yml:1]: oh did not know it was changed. good to know, thanks for the pointer

**Review by epinault — APPROVED** [review]

**Comment by epinault** [comment]
what s the plan here? merge in master or are we gonna first release Hammer then update this to proper version and update? this is still pointing to a branch

**Comment by ruslandoga** [comment]
The Hammer PR needs to be merged first. Then the backends. I described the plan in the Hammer PR.

File: lib/hammer/redis.ex [hunk 2]
```diff
@@ -51,6 +53,7 @@ defmodule Hammer.Redis do
       @prefix unquote(prefix)
       @timeout unquote(timeout)
 
+      @spec child_spec(Keyword.t()) :: Supervisor.child_spec()
```
Comment (ruslandoga) [inline lib/hammer/redis.ex:57]: Why is it `Keyword.t` here but `Redis.redis_opts()` in `start_link`? :)

File: lib/hammer/redis.ex [hunk 3]
```diff
@@ -59,30 +62,36 @@ defmodule Hammer.Redis do
         }
       end
 
+      @spec start_link(Hammer.Redis.redis_opts()) :: {:ok, Redix.connection()}
```
Comment (ruslandoga) [inline lib/hammer/redis.ex:66]: This type is not really valid.

File: lib/hammer/redis.ex [hunk 4]
```diff
@@ -59,30 +62,36 @@ defmodule Hammer.Redis do
...
       def start_link(opts) do
         opts = Keyword.put(opts, :name, @name)
         Hammer.Redis.start_link(opts)
       end
 
-      def hit(key, scale, limit, increment \\ 1, timeout \\ @timeout) do
-        Hammer.Redis.hit(@name, @prefix, key, scale, limit, increment, timeout)
```
Comment (ruslandoga) [inline lib/hammer/redis.ex]: Please remove the timeout arg. It was by design.

File: lib/hammer/redis.ex [hunk 5]
```diff
@@ -59,30 +62,36 @@ defmodule Hammer.Redis do
...
+      @impl Hammer
```
Comment (ruslandoga) [inline lib/hammer/redis.ex:72]: `@impl` here is unnecessary.

**Comment by ruslandoga** [comment]
👋 @epinault 

This PR was also merged prematurely :)

Please ask me first before merging my work. And also please stop adding commits to my branches. If you don't think we can work together like this, then I'll have to stop with my contributions.

**Comment by epinault** [comment]
As mentioned in other PR, I am happy to chat with you so we can figure a way to collaborate. there was no intention to make you feel this way. As maintainer, i am just trying to move the needle on this package as. I have a little more time lately.

File: lib/hammer/redis.ex (hunk 5 above)
Comment (epinault) [reply lib/hammer/redis.ex:72]: interesting , because it s in a macros it s not needed. TIL

File: lib/hammer/redis.ex (hunk 4 above)
Comment (epinault) [reply lib/hammer/redis.ex]: that seems very confusing to have a callbacks /3 and /4 and a call here. /5 that don match.  It feels as use a little confusing. I had it separate in prior commit but it felt this is something we could do later.  I thought we could start matching the contract first and seprate the functional part already and add this later. Should have asked you with retrospect.

File: lib/hammer/redis.ex (hunk 3 above)
Comment (epinault) [reply lib/hammer/redis.ex:66]: interesting. AI failed, Dialyzer did not pick it up.

**Pull request merged**